https.go

```diff
@@ -555,8 +555,10 @@ func httpsProxyFromEnv(reqURL *url.URL) (string, error) {
 
 	service := proxyURL.Port()
 	if service == "" {
+		// TODO I am very confused by this line of code
+		// what does a URL of the form `proxy-example.com:http` mean??
 		service = proxyURL.Scheme
 	}
 
-	return fmt.Sprintf("%s:%s", proxyURL.Hostname(), service), nil
+	return fmt.Sprintf("//%s:%s", proxyURL.Hostname(), service), nil
```

Comment:
Assuming you agree with my comments above, we should remove the `//` prefix since this will no longer be passed to `url.Parse`.